sdk/iot/hub/src/az_iot_hub_client_sas.c

```diff
@@ -14,110 +14,165 @@
 #define LF '\n'
 #define AMPERSAND '&'
 #define EQUAL_SIGN '='
+#define STRING_NULL_TERMINATOR '\0'
 #define SCOPE_DEVICES_STRING "/devices/"
+#define SCOPE_MODULES_STRING "/modules/"
 #define SAS_TOKEN_SR "SharedAccessSignature sr"
 #define SAS_TOKEN_SE "se"
 #define SAS_TOKEN_SIG "sig"
 #define SAS_TOKEN_SKN "skn"
 
-az_result az_iot_sas_token_get_document(
-    az_span iothub_fqdn,
-    az_span device_id,
-    int32_t expiry_time_secs,
-    az_span document,
-    az_span* out_document)
+static const az_span devices_string = AZ_SPAN_LITERAL_FROM_STR(SCOPE_DEVICES_STRING);
+static const az_span modules_string = AZ_SPAN_LITERAL_FROM_STR(SCOPE_MODULES_STRING);
+static const az_span skn_string = AZ_SPAN_LITERAL_FROM_STR(SAS_TOKEN_SKN);
+static const az_span sr_string = AZ_SPAN_LITERAL_FROM_STR(SAS_TOKEN_SR);
+static const az_span sig_string = AZ_SPAN_LITERAL_FROM_STR(SAS_TOKEN_SIG);
+static const az_span se_string = AZ_SPAN_LITERAL_FROM_STR(SAS_TOKEN_SE);
+
+AZ_NODISCARD az_result az_iot_hub_client_sas_get_signature(
+    az_iot_hub_client const* client,
+    uint32_t token_expiration_epoch_time,
+    az_span signature,
+    az_span* out_signature)
 {
-  AZ_PRECONDITION_VALID_SPAN(device_id, 1, false);
-  AZ_PRECONDITION_VALID_SPAN(iothub_fqdn, 1, false);
-  AZ_PRECONDITION_VALID_SPAN(document, 0, false);
-  AZ_PRECONDITION_NOT_NULL(out_document);
+  AZ_PRECONDITION_NOT_NULL(client);
+  AZ_PRECONDITION(token_expiration_epoch_time > 0);
+  AZ_PRECONDITION_VALID_SPAN(signature, 1, false);
+  AZ_PRECONDITION_NOT_NULL(out_signature);
 
-  az_span devices_string = AZ_SPAN_FROM_STR(SCOPE_DEVICES_STRING);
-  int32_t required_length
-      = az_span_size(iothub_fqdn) + az_span_size(device_id) + az_span_size(devices_string) + 1;
+  int32_t required_size = az_span_size(client->_internal.iot_hub_hostname)
+      + az_span_size(devices_string)
+      + az_span_size(client->_internal.device_id)
+      + (int32_t)sizeof(LF)
+      + u32toa_size(token_expiration_epoch_time);
 
-  AZ_RETURN_IF_NOT_ENOUGH_SIZE(document, required_length);
+  if (az_span_size(client->_internal.options.module_id) > 0)
+  {
+    required_size
+        += + az_span_size(modules_string) + az_span_size(client->_internal.options.module_id);
```

Comment:
>+ [](start = 11, length = 1)

+?

Reply:
yeah, I think I saw other comments before of people in our team preferring the explicit if block. I know it's bigger, but more people prefer it. Sorry.